### ISSUE TITLE
Missing Input Options for Activity Log and Diagnostics

### ISSUE THREAD

**Issue by mary-cordova** [microsoft/AzureMonitorAddonForSplunk#17]
Perhaps there was something wrong with my install?  (Also, I am running Splunk as not root, I performed all (Linux) installation steps as root but chowned the directory to the splunk user and restarted Splunk.)

The only Data Input option available is "Azure Monitor Metrics", but the one I want most to start with is "Azure Monitor Activity Log"

![missing_inputs](https://user-images.githubusercontent.com/17890064/35420233-36fa200e-01f1-11e8-8b8e-e3d556af0205.PNG)

**Comment by mary-cordova** [comment]
Not sure if this is an issue or not:

root@AW-SK-WEBF-P-01:/opt/splunk/etc/apps/TA-Azure_Monitor/bin/app# npm install
npm WARN azure_monitor@1.2.0 No repository field.

**Comment by mary-cordova** [comment]
I've tried installing several versions of the .spl (though I cannot confirm I've attempted every version) with the same result.  

I'm running Ubuntu 16 but have also tried on Ubuntu 14.

**Comment by mary-cordova** [comment]
I manually created the Activity Log input.conf but I'm not sure this works, I've got an event hub setup in Azure and it looks like data is being sent to it.  I cannot tell from the Azure portal if my Splunk configuration is making successful calls.  

![manual_input](https://user-images.githubusercontent.com/17890064/35643787-27a1bbe6-067c-11e8-8eb7-9d6ea5729c86.PNG)

**Comment by iainelliott** [comment]
Mary, did you work out why this was happening? I have exactly the same issue.

**Comment by iainelliott** [comment]
Hi Mary, did you just add that to the end of input.conf in the default folder?
The npm error means nothing, its a text field description thats missing, very common.

**Comment by mary-cordova** [comment]
i made a copy of the inputs.conf.spec file thats in the readme dir but its not working

i have a heavy forwarder on windows with the webui enabled, im going to install on that and see if it's any different

**Comment by mary-cordova** [comment]
Windows install also missing inputs for Activity and Diagnostic.
![windows_install_missing_input](https://user-images.githubusercontent.com/17890064/35691430-52aa8dfc-072d-11e8-996d-e62460b20b1b.PNG)

**Comment by mary-cordova** [comment]
another test im running, i created the input via the webui for monitor metrics then modified/added the bold items the resulting input (under the launcher app):  

[**azure_activity_log**://Azure Activity IRM]
SPNApplicationId = ********
SPNApplicationKey = ********
SPNTenantID = string
SubscriptionId = string
index = azure
interval = 60
secretName = string
secretVersion = string
vaultName = string
**eventHubNamespace = string**
sourcetype = **string**

no  luck so far but maybe i need to wait a bit for things to start coming through

**Comment by iainelliott** [comment]
I created a folder in the app directory called 'local' and then create a file inside that called 'inputs.conf',  if you put the following in then it should configure the inputs:

[azure_diagnostic_logs://<value>]
SPNTenantID = <value>
SPNApplicationId = <value>
SPNApplicationKey = <value>
eventHubNamespace = <value>
vaultName = <value>
secretName = <value>
secretVersion = <value>
index = main
interval = 60
sourcetype = amdl:diagnosticLogs

[azure_activity_log://<value>]
SPNTenantID = <value>
SPNApplicationId = <value>
SPNApplicationKey = <value>
eventHubNamespace = <value>
vaultName = <value>
secretName = <value>
secretVersion = <value>
index = main
interval = 60
sourcetype = amal:activityLog

[azure_monitor_metrics://<name>]
SPNTenantID = <value>
SPNApplicationId = ********
SPNApplicationKey = ********
SubscriptionId = <value>
vaultName = <value>
secretName = <value>
secretVersion = <value>
index = main
interval = 60
sourcetype = amm:metrics

I'm still confused by the 'Unable to initialize modular input' error, it feels like there are files missing somewhere as all the modular inputs are not defined.

Try connecting to: https://<your server>:8089/services/data/modular-inputs/azure_monitor_metrics and you'll see that only the monitor metrics one appears. I might be barking up the wrong tree but thats where I got to with it.

**Comment by iainelliott** [comment]
As root
cd /opt/splunk/etc/app/TA-Azure_Monitor/bin
npm install node-sass request@2.81.0
reboot and the error goes away.

Found the solution here: https://github.com/sass/node-sass/issues/2100

**Comment by mary-cordova** [comment]
so you got all 3 inputs working now?!  ill try it :)

**Comment by danielwnn** [comment]
I got all 3 data input sources, configured the Activity Log, but no data coming from Even Hub. How to debug? Thanks!

**Comment by mary-cordova** [comment]
Yeah I have the same issue, finally got all the sources to show up after
the other poster's tips but no data either.

I've been on top of our Microsoft and Splunk support contacts and have
calls to troubleshoot with them next week.

I'll update this after.

**Comment by mary-cordova** [comment]
ok so i talked to someone from microsoft, there is a fork for this, greg is not really supporting it anymore but rick is: ttps://github.com/rickrain/AzureMonitorAddonForSplunk

rick said the powershell script he has is handy and works

i tried deploying again and running as root and that did not fix the issue

HOWEVER, I deployed this in Azure using the Azure VM version of Splunk Enterprise and it works (or at least I have all the inputs available - still using the installation instructions from the previous version and dependency script from the pull request there also).

I noticed that the Azure Splunk VM is running Splunk version 6.6.1 - I would venture to guess that an on-prem installation using the same Splunk version would also work...I will test that out tomorrow.  

I have not yet hooked this up to the Azure pieces (Event Hubetc).

**Comment by iainelliott** [comment]
Awesome Mary, great to know that there is a fork and someone is looking after it at least intermittently.

**Comment by mary-cordova** [comment]
upgraded 1 server to splunk version 7.0.2 and i now have all 3 inputs available

i have not yet configured

![splunk_702](https://user-images.githubusercontent.com/17890064/36223218-9015b9fa-1178-11e8-9ecc-f40af63a29c4.PNG)

**Comment by mary-cordova** [comment]
ok, im running this locally and in an azure marketplace instance of splunk enterprise and have it configured for activity logs but am still not seeing any data in splunk

**Comment by ielliott75** [comment]
We have moved away from this as it appears that MS has changed functionality in event hub without maintaining backwards compatibility. If you are looking to build an enterprise monitoring solution I can’t see how you can use this with any certainty. Azure monitor service (part of Azure) seems to provide the same functionality.

**Comment by OzeroLee** [comment]
@mary-cordova Were you able to get data from Azure? We are having a similar issue.

**Comment by mary-cordova** [comment]
here is a better way to do this without using any apps :D

https://answers.splunk.com/answers/678660/how-to-get-logs-from-azure-and-o365-into-splunk.html

**Comment by sebastus** [comment]
Hi Mary. Thanks for your note. There is yet another way. Installation is a breeze. https://github.com/Microsoft/AzureFunctionforSplunkVS. 

On the topic of not getting data from Azure Monitor via the TA. I recently spoke at length to a customer. His issue was that he had not set up one of the service principals correctly. This is a pretty popular one.

**Comment by pelsebubb** [comment]
Hi.  I have the same problem using MSI auth.  I get no data into splunk from eventhubs.  Is this addon planned to be maintaned or should we use functions to ship data ?

**Comment by mary-cordova** [comment]
I think this has effectively been abandoned, I know there's a fork of it that was being maintained by someone else, but that was 6+ months ago so no idea if it's still maintained.  Also, there's no mention of it but this only works on an Azure marketplace instance of Splunk Enterprise.  I did get the fork working on an Azure deployment that I then forwarded to my on-prem deployment but it was still an unsupported flaky black-box...none of which are things I like.

I built my own solution in Azure with "Logic Apps": https://answers.splunk.com/answers/678660/how-to-get-logs-from-azure-and-o365-into-splunk.html

It's simple and has been pretty reliable (tho it is delayed by 5 hours...I think I could get it down to 3...but it's working so I haven't messed with it).
